CMakeLists.txt

```diff
@@ -2,51 +2,100 @@ project(libMCNP2CAD)
-
-find_package(Eigen3 3.3 REQUIRED NO_MODULE)
-
-# These directories need to be defined by the user.
-include_directories(${IGEOM_INCLUDE_DIR})
-find_library(IGEOM_LIB NAMES iGeom HINTS ${IGEOM_LIB_DIR})
-
-SET(LIB_SRC
-    mcnp2cad.cpp
-    mcnp2cad.hpp
-    MCNPInput.cpp
-    MCNPInput.hpp
-    geometry.cpp
-    geometry.hpp
-    volumes.cpp
-    volumes.hpp
-    GQ_Characterize.cpp
-    GQ_Characterize.hpp
-    test_GQ.cpp
-)
-
-add_library(mcnp2cad SHARED ${LIB_SRC})
-target_link_libraries(mcnp2cad ${IGEOM_LIB})
-
-add_executable(test_GQ test_GQ.cpp GQ_Characterize.cpp)
-target_link_libraries(test_GQ)
-if(IGEOM_LIB_DIR)
-    set_target_properties(test_GQ PROPERTIES INSTALL_RPATH ${IGEOM_LIB_DIR} BUILD_WITH_INSTALL_RPATH TRUE)
-elseif(CMAKE_PREFIX_PATH)
-    set_target_properties(test_GQ PROPERTIES INSTALL_RPATH ${CMAKE_PREFIX_PATH} BUILD_WITH_INSTALL_RPATH TRUE)
-endif()
-
-enable_testing()
-add_test(NAME test_gq_characterization COMMAND test_GQ)
+    endif ()
+  endif ()
+endif ()
+
+# Find iGeom. If mcnp2cad is being built as a standalone library, then
+# -DIGEOM_DIR must be specified. If it is not being built as a standalone
+# library, then it is assumed that the location of the iGeom library is known
+# and that the iGeom headers have already been included.
+if (STANDALONE_MCNP2CAD)
+  if (NOT IGEOM_DIR)
+    message(FATAL_ERROR "-DIGEOM_DIR must be specified when building mcnp2cad"
+                        " as a standalone library")
+  else ()
+    message(STATUS "Building standalone mcnp2cad")
+  endif ()
+
+  # Find location of iGeom headers
+  find_path(IGEOM_INCLUDE_DIR NAMES iGeom.h HINTS ${IGEOM_DIR}/include)
+  include_directories(${IGEOM_DIR}/include)
+  message(STATUS "IGEOM_INCLUDE_DIR: ${IGEOM_INCLUDE_DIR}")
+
+  # Find location of iGeom library
+  find_library(IGEOM_LIB NAMES iGeom HINTS ${IGEOM_DIR}/lib)
+  get_filename_component(IGEOM_LIB_DIR ${IGEOM_LIB} DIRECTORY)
+  link_directories(${IGEOM_LIB_DIR})
+  message(STATUS "IGEOM_LIB_DIR: ${IGEOM_LIB_DIR}")
+else ()
+  message(STATUS "Building mcnp2cad as a component of another package")
+endif ()
+
+# RPATH handling. If CMAKE_INSTALL_RPATH_USE_LINK_PATH is set to ON, then all
+# directories that have been linked via the link_directories() command will be
+# added to the RPATH.
+if (STANDALONE_MCNP2CAD)
+  set(CMAKE_INSTALL_RPATH_USE_LINK_PATH ON)
+endif ()
+
+SET(MCNP2CAD_SRC_FILES geometry.cpp
+                       GQ_Characterize.cpp
+                       mcnp2cad.cpp
+                       MCNPInput.cpp
+                       volumes.cpp)
+
+set(MCNP2CAD_PUB_HEADERS dataref.hpp
+                         geometry.hpp
+                         GQ_Characterize.hpp
+                         mcnp2cad.hpp
+                         MCNPInput.hpp
+                         options.hpp
+                         version.hpp
+                         volumes.hpp)
+
+# mcnp2cad library
+add_library(mcnp2cad SHARED ${MCNP2CAD_SRC_FILES})
+target_link_libraries(mcnp2cad iGeom)
```

Comment:
Rather than link to the library name and expect it to be in the set of link directories, let's rely the `IGEOM_LIB` target created by the `find_library` call above.

Reply:
Done